tests/Microsoft.Identity.Test.Core.UIAutomation/MobileTestHelper.cs

```diff
@@ -280,10 +280,12 @@ public void VerifyResult(ITestController controller)
                 }
                 else if (result.Contains(CoreUiTestConstants.TestResultFailureMessage))
                 {
+                    controller.Tap("LogPage");
```

Comment:
You don't need all those `else` clauses because you are either returning or throwing, which breaks the flow anyway.